tinygrad/codegen/uopgraph.py

```diff
@@ -543,8 +546,9 @@ def linearize(self, extra_pm:Optional[PatternMatcher]=None, skip_check=False) ->
     # filter nodes that don't link to a sink
     # BFS toposort
     children: Dict[UOp, List[UOp]] = {}
+    range_srcs: Dict[UOp, Set[UOp]] = {}
```

Comment:
so arg[1] refers to if the loop is implicit (gidx) or explicit (for) - since we don't care about the implicit ones already I think this can be faster if we just track `arg[1] is True`

Reply:
what do you mean by just tracking `arg[1] is True`? I think it already does check `arg[1]` before adding to `range_srcs`

I need to keep some unique identifier so that we can prioritize ranges by how deep they are in the ast, if we want to be extra tight on space I could probably track `arg[0]`

Reply:
I see a few checks for truthy `arg[1]` in the lines after this, why do you need to track the global loops?

Reply:
I think they're leftover from debugging, I should be able to get rid of them